presto-orc/src/main/java/com/facebook/presto/orc/writer/SliceDictionaryColumnWriter.java

```diff
@@ -222,8 +220,15 @@ private void writeDictionaryRowGroup(Block dictionary, int valueCount, IntBigArr
             DictionaryBlock dictionaryBlock = new DictionaryBlock(positionCount, dictionary, segment);
             directColumnWriter.writeBlock(dictionaryBlock);
             valueCount -= positionCount;
+
+            if (directColumnWriter.getBufferedBytes() > maxDirectBytes) {
+                directColumnWriter.close();
```

Comment:
Just curious, why move these here.  I ask because the caller is still doing `directColumnWriter.finishRowGroup();`

Reply:
Good point, I originally want to avoid repeating the following two lines:

```java
    directColumnWriter.close();
    directColumnWriter.reset();
```

But it makes the abstraction a bit weird. As you pointed out `finishRowGroup` is still done by callee.